docs/references/contributor_guide.md

```diff
@@ -1,22 +1,126 @@
 
-## For Newcomers
-If you want to contribute or learn but don't have a concrete idea yet, you can pick a task labeled as "good first issue" or "help wanted" from the list below.
-[https://github.com/sgl-project/sglang/issues?q=is%3Aissue+label%3A%22good+first+issue%22%2C%22help+wanted%22](https://github.com/sgl-project/sglang/issues?q=is%3Aissue+label%3A%22good+first+issue%22%2C%22help+wanted%22)
+- **pre-commit install**: Installs the Git hook, which automatically runs style checks on staged files.
+- **pre-commit run --all-files**: Manually runs all the configured checks on your repository, applying automatic fixes where possible.
+
+Please ensure your code passes all these checks locally before creating a Pull Request.
+
+## 3. Writing Documentation & Running Docs CI
+
+All documentation files are located in the docs folder. We use make commands for building, cleaning, and previewing the documentation. Below is the typical workflow:
+
+1. **Compile Jupyter Notebooks**
+
+bash
+   make compile
+
+   This command processes all .ipynb files into intermediate formats.
+
+2. **Generate HTML Documentation**
+
+bash
+   make html
+
+   The generated static site can be found under the _build/html directory.
+
+3. **Local Preview**
+
+bash
+   bash serve.sh
+
+   Open your browser at the specified port to view the docs locally.
+
+4. **Clean Notebook Outputs & Build Artifacts**
+
+bash
+   find . -name '*.ipynb' -exec nbstripout {} \;
+   make clean
+
+   - **nbstripout** removes all outputs from Jupyter notebooks.
+   - **make clean** removes temporary files and _build artifacts.
+
+5. **Modify index.rst (If Necessary)**
+   Update index.rst whenever you add or remove sections from the documentation.
+
+6. **Pass Pre-Commit and Submit a PR**
+   After confirming the docs build successfully and pass all checks, open a Pull Request.
+
+CI will automatically build and validate the documentation to ensure consistency and completeness before merging.
+
+## 4. Running Unit Tests Locally & Adding to CI
+
+SGLang uses Python’s built-in [unittest](https://docs.python.org/3/library/unittest.html) framework. You can run tests either individually or in suites.
+
+### 4.1 Test Backend Runtime
+
+bash
+cd sglang/test/srt
+
+# Run a single file
+python3 test_srt_endpoint.py
+
+# Run a single test
+python3 -m unittest test_srt_endpoint.TestSRTEndpoint.test_simple_decode
+
+# Run a suite with multiple files
+python3 run_suite.py --suite minimal
+
+
+### 4.2 Test Frontend Language
+
+bash
+cd sglang/test/lang
+export OPENAI_API_KEY=sk-*****
+
+# Run a single file
+python3 test_openai_backend.py
+
+# Run a single test
+python3 -m unittest test_openai_backend.TestOpenAIBackend.test_few_shot_qa
+
+# Run a suite with multiple files
+python3 run_suite.py --suite minimal
+
+
+### 4.3 Adding or Updating Tests in CI
+
+- Create new test files under test/srt, ensure they are referenced in test/srt/run_suite.py, or they will not be picked up.
```

Comment:
- Create new test files under `/test`, and ensure they are referenced in `/test/srt/run_suite.py` or `/test/lang/run_suite.py`, other they will not be picked up in CI.